Model/Feed/Runner.php

```diff
@@ -182,12 +203,22 @@ public function doFeed(array $feedTypes, int $storeId): void
      */
     public function sendFeed(int $storeId, string $type) : void
     {
+        $feedHandler = $this->feedTypeHandler->getFeedHandler($type);
         try {
-            $feedHandler = $this->feedTypeHandler->getFeedHandler($type);
             if ($feedHandler->isEnabled($storeId)) {
-                $this->handleFeed($feedHandler, $storeId, $type);
+                $store = $this->getStore($storeId);
+                if ($feedHandler->requiresEmulation()) {
+                    $this->appEmulation->startEnvironmentEmulation($storeId, Area::AREA_FRONTEND, true);
+                }
+                $this->handleFeed($feedHandler, $store, $type);
+                if ($feedHandler->requiresEmulation()) {
+                    $this->appEmulation->stopEnvironmentEmulation();
+                }
             }
         } catch (Exception $e) {
+            if ($feedHandler->requiresEmulation()) {
+                $this->appEmulation->stopEnvironmentEmulation();
```

Comment:
I assume this just works if we hit the catch prior to starting an emulation? As in it won't error because there's no emulation to stop?

Reply:
yep, the stopEnvironmentEmulation method itself handles whether emulation has actually started or not so won't try to stop emulation that isn't started :)